Source/AppAuthTV/OIDTVAuthorizationService.m

```diff
@@ -39,8 +41,94 @@
  */
 NSString *const kErrorCodeSlowDown = @"slow_down";
 
+/*! @brief Path appended to an OpenID Connect issuer for discovery
+    @see https://openid.net/specs/openid-connect-discovery-1_0.html#ProviderConfig
+ */
+static NSString *const kOpenIDConfigurationWellKnownPath = @".well-known/openid-configuration";
+
 @implementation OIDTVAuthorizationService
 
+#pragma mark OIDC Discovery
+
++ (void)discoverServiceConfigurationForIssuer:(NSURL *)issuerURL
+                                   completion:(OIDTVDiscoveryCallback)completion {
+  NSURL *fullDiscoveryURL =
+      [issuerURL URLByAppendingPathComponent:kOpenIDConfigurationWellKnownPath];
+
+  [[self class] discoverServiceConfigurationForDiscoveryURL:fullDiscoveryURL
+                                                 completion:completion];
+}
+
++ (void)discoverServiceConfigurationForDiscoveryURL:(NSURL *)discoveryURL
```

Comment:
Seems like there's a lot of shared code with OIDAuthorizationService's method. I think there's an opportunity to rewrite the classes/methods to reduce duplication.

(This is not an actionable item, just general musing).

Reply:
I reduced the duplication by having [discoverServiceConfigurationForDiscoveryURL:] call the function in OIDAuthorizationService